gui/src/controls/button.ts

```diff
@@ -136,5 +136,23 @@ module BABYLON.GUI {
 
             return result;
         }
+        
+        public static CreateImageWithCenterTextButton(name: string, text: string, imageUrl: string): Button {
+            var result = new Button(name);
+
+            // Adding text
+            var textBlock = new BABYLON.GUI.TextBlock(name + "_button", text);
+            textBlock.textWrapping = true;
+            textBlock.zIndex = 10; 
+            textBlock.textHorizontalAlignment = BABYLON.GUI.Control.HORIZONTAL_ALIGNMENT_CENTER;
+            result.addControl(textBlock);   
+
+            // Adding image
+            var iconImage = new BABYLON.GUI.Image(name + "_icon", imageUrl);
+            iconImage.stretch = BABYLON.GUI.Image.STRETCH_FILL;

```

Comment:
Should the image be behind the text?

Reply:
Correct, the image is a background.
Here's an example which isn't grass :)
https://raw.githubusercontent.com/aWeirdo/Babylon.js/master/button.png

Reply:
So you should add the image first

Reply:
i'll swap them around if you prefer.
i just did it with zIndex as i liked keeping the textPanel as the first child.

Reply:
The zindex will reord the children

Reply:
Ahh, i see, when the control is added it..